contracts/modules/PermissionManager/GeneralPermissionManager.sol

```diff
@@ -101,4 +99,115 @@ contract GeneralPermissionManager is IPermissionManager, Module {
         return allPermissions;
     }
 
+    /**
+    * @notice use to get all delegates
+    * @return address[]
+    */
+    function getAllDelegates() public view returns(address[]) {
+        return allDelegates;
+    }
+
+    /**
+    * @notice use to check if an address is a delegate or not
+    * @param _potentialDelegate the address of potential delegate
+    * @return bool
+    */
+    function isDelegate(address _potentialDelegate) public view returns(bool) {
+
+        require(_potentialDelegate != address(0));
+
+        if (delegateDetails[_potentialDelegate] != bytes32(0)) {
+            return true;
+        }else{
+            return false;
+        }
+    }
+
+    /**
+    * @notice Use to change one or more permissions for a single delegate at once
+    * @param _delegate Ethereum address of the delegate
+    * @param _multiModules Multiple module matching the multiperms, needs to be same length
+    * @param _multiPerms Multiple permission flag needs to be changed
+    * @return nothing
+    */
+    function changePermissionBulk(
+        address _delegate,
+        address[] _multiModules,
+        bytes32[] _multiPerms
+    )
+    external
+    withPerm(CHANGE_PERMISSION)
+    {
+        require(delegateDetails[_delegate] != bytes32(0), "Delegate details not set");
+        require(_multiModules.length != 0 && _multiPerms.length !=0);
+        
+        for(uint8 i=0;i<_multiPerms.length;i++){
+            bool _currentPerm = !perms[_multiModules[i]][_delegate][_multiPerms[i]];
+            perms[_multiModules[i]][_delegate][_multiPerms[i]] = _currentPerm;
+            emit ChangePermission(_delegate, _multiModules[i], _multiPerms[i], _currentPerm, now);
+        }
+    }
+
+    /**
+    * @notice use to return all delegates with a given permission and module
+    * @param _module Ethereum contract address of the module
+    * @param _perm Permission flag
+    * @return address[]
+    */
+
+    address[] allDelegatesWithPerm;
+
+    function getAllDelegatesWithPerm(address _module, bytes32 _perm) public returns(address[]) {
+
+        require(_module != address(0) && _perm != bytes32(0));
+
+        for(uint8 i=0;i<allDelegates.length;i++){
+            if (perms[_module][allDelegates[i]][_perm]){
+                allDelegatesWithPerm.push(allDelegates[i]);
+            } else {}
+        }
+
+        return allDelegatesWithPerm;
+    }
+
+    /**
+    * @notice use to return all permission of a signle or multiple module
+    * @param _delegate Ethereum address of the delegate
+    * @param _tokenAddress Ethereum address of the security token
+    * @param _types uint8[] of types
+    * @return address[] the address array of Modules this delegate has permission
+    * @return bytes32[] the permission array of the corresponding Modules
+    */
+
+    address[] _allModules;
+    bytes32[] _allPerms;
+
```

Comment:
same here. Temporary type variables should be declared within the function of the contract